tests/system_tests_two_routers.py

```diff
@@ -409,6 +409,14 @@ def __init__(self, address):
         self.mgmt_sender = None
         self.success = False
         self.error = None
+        self.receiver_to_kill = None
+        self.timer = None
+        self.n_sent = 0
+        self.n_received = 0
+        self.mgmt_receiver_link_opened = False
+        self.mgmt_receiver_1_link_opened = False
+        self.mgmt_receiver_2_link_opened = False
+        self.query_timer = None
 
     def on_start(self, event):
         self.timer = event.reactor.schedule(TIMEOUT, TestTimeout(self))
```

Comment:
Sorry but github won't let me comment on line 438:
The timeout(self) method should not simply close self.mgmt_conn - it should call self.bail() with the given error message